code/modules/mob/living/carbon/human/species/station/prommie_blob.dm

```diff
@@ -0,0 +1,466 @@
+
+	//Put our owner in it (don't transfer var/mind)
+	qdel(blob.ai_holder)
+
+	blob.ai_holder = null
+	blob.ckey = ckey
+	blob.name = name
+	blob.nutrition = nutrition
+	blob.color = rgb(r_skin, g_skin, b_skin)
+	blob.mood = ":3"
+	if(radiation > 0)
+		blob.rad_glow = CLAMP(radiation,0,250)
+		set_light(0)
+		blob.set_light(max(1,min(5,radiation/15)), max(1,min(10,radiation/25)), blob.color)
+		blob.handle_light()
+	if(has_hat)
+		blob.hat = new_hat
+		new_hat.forceMove(src)
+	blob.update_icon()
+	blob.verbs -= /mob/living/proc/ventcrawl // Absolutely not.
+	blob.verbs -= /mob/living/simple_mob/proc/set_name // We already have a name.
+	temporary_form = blob
+
+	//Mail them to nullspace
+	moveToNullspace()
+
+	//Message
+	blob.visible_message("<b>[src.name]</b> squishes into their true form!")
+
+	//Transfer vore organs
+	blob.vore_organs = vore_organs
+	blob.vore_selected = vore_selected
+	for(var/obj/belly/B as anything in vore_organs)
+		B.forceMove(blob)
+		B.owner = blob
+
+	//We can still speak our languages!
+	blob.languages = languages.Copy()
+
+	//Return our blob in case someone wants it
+	return blob
+
+/mob/living/carbon/human/proc/prommie_outofblob(var/mob/living/simple_mob/slime/promethean/blob, force)
+	if(!istype(blob))
+		return
+
+	if(!force && !isturf(blob.loc))
+		to_chat(blob,"<span class='warning'>You can't change forms while inside something.</span>")
+		return
+
+	if(buckled)
+		buckled.unbuckle_mob()
+	if(LAZYLEN(buckled_mobs))
+		for(var/buckledmob in buckled_mobs)
+			riding_datum.force_dismount(buckledmob)
+	if(pulledby)
+		pulledby.stop_pulling()
+	stop_pulling()
+
+	//Message
+	blob.visible_message("<b>[src.name]</b> pulls together, forming a humanoid shape!")
+
+	//Record where they should go
+	var/atom/reform_spot = blob.drop_location()
+
+	//Size update
+	resize(blob.size_multiplier, FALSE, ignore_prefs = TRUE)
+
+	//Move them back where the blob was
+	forceMove(reform_spot)
+
+	//Put our owner in it (don't transfer var/mind)
+	ckey = blob.ckey
+	var/obj/item/hat = blob.hat
+	blob.drop_hat()
+	drop_from_inventory(hat) // Hat me baby
+	equip_to_slot_if_possible(hat, slot_head)
+	nutrition = blob.nutrition // food good
+
+
+	temporary_form = null
+
+	shapeshifter_set_colour(blob.color)
+	if(blob.rad_glow > 0)
+		radiation = CLAMP(blob.rad_glow,0,250)
+		set_light(max(1,min(5,radiation/15)), max(1,min(10,radiation/25)), species.get_flesh_colour(src))
+	update_icon()
+
+	//Transfer vore organs
+	vore_selected = blob.vore_selected
+	for(var/obj/belly/B as anything in blob.vore_organs)
+		B.forceMove(src)
+		B.owner = src
+
+	//vore_organs.Cut()
+	if(blob.prev_left_hand) put_in_l_hand(blob.prev_left_hand) //The restore for when reforming.
+	if(blob.prev_right_hand) put_in_r_hand(blob.prev_right_hand)
+
+	Life(1) //Fix my blindness right meow //Has to be moved up here, there exists a circumstance where blob could be deleted without vore organs moving right.
+
+	//Get rid of friend blob
+	qdel(blob)
```

Comment:
It would probably also be good if we didn't delete the blob, and instead maybe stored it in the prommie body. Unless I'm reading things wrong, it seems like you'd need to redo any customization every time you flip flopped between them. Just as a suggestion~

Reply:
Aye, on it

Reply:
I think easier is middle ground ,where just customization is saved as vars on species datum or something and re-applied when you return to blobform. Proteans delete their blobform too because of technical fangaling. Just a suggestion here too though.